File: C#/Lista 1/02.cs
/* Adaptar o programa acima para que ele calcule o percentual dos valores positivos,
negativos e zeros em relação ao total de valores fornecidos. */

using System;

class Program {
  public static void Main (string[] args) {
    int[] numeros; //aramzenar sequencia de numeros
    int tam = 0;

    Console.WriteLine("Digite a quantidade de numeros: ");
    tam = int.Parse(Console.ReadLine()); // ler quantidade de numeros da sequencia
            
    numeros = new int[tam];
            
    Console.WriteLine("Digite a sequencia de numeros: ");
    for(int i = 0; i < tam; i++){
        numeros[i] = int.Parse(Console.ReadLine()); //armazenar os numeros em cada posicao do vetor 
    }
       
    int pos = 0, neg = 0, zero = 0; //variaveis para fazer a contagem 

    for(int i = 0; i < tam ; i++){
        if(numeros[i] > 0){ //verifica se o numero e positivo
            pos++;
        }else if(numeros[i] < 0){ //verifica se o numero e negativo
            neg++;
        }else{ // qunado o numero nn se encaixa em nenhuma das verificações
            zero++;
        }
    }

    //calcular o percentual
    percPos = (pos*100)/tam; //positivo
    percNeg = (neg*100)/tam; //negativo
    percZero = (zero*100)/tam; //zeros

    Console.WriteLine("A quamtidade de numeros positivos é " + pos + " , de negativos é " + neg + " e de zeros é " + zero); // imprimir quantidades
    Console.WriteLine("A porcentagem de positivos é " + percPos + " , de negativos é " + percNeg + " e de zeros é " + percZero); // imprimir resultados
  }
}

File: C/Lista 1/02.c
/* Adaptar o programa acima para que ele calcule o percentual dos valores positivos,
negativos e zeros em relação ao total de valores fornecidos. */

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

/*----- Função main ------*/
int main(){

    //saber o tamanho da sequencia
    int tam;
    printf("\nDigite o tamanho da sequencia de numeros: ");
    scanf("%i",&tam);

    //declarar variaveis
    int num[tam];
    int pos = 0;
    int neg = 0;
    int zero = 0;
    float porctPos = 0;
    float porctNeg = 0;
    float porctZero = 0;

    printf("\nDigite uma sequencia de numeros:\n");
    for(int i = 0; i < tam; i++){ //loop para adicionar os valores ao vetor
        scanf(" %i", &num[i]);
    }

    for(int j = 0; j < tam; j++){ //loop para percorrer os valores do vetor
        if(num[j] > 0){ //verifica se o numero e positivo
            pos++;
        }else if(num[j] < 0){ //verifica se o numero e negativo
            neg++;
        }else{ //caso o numero seja zero
            zero++;
        }
    }

    porctPos = (pos * 100)/tam;
    porctNeg = (neg * 100)/tam;
    porctZero = (zero * 100)/tam; 

    printf("\n------- Resutados -------");
    printf("\nNa sequencia ha:\n%.1f%% de numeros positivos\n%.1f%% de numeros negativos\n%.1f%%  de numeros zeros.", porctPos, porctNeg, porctZero);

    return 0;

}